workspaces/cli-server/src/routers/capture-router.ts

```diff
@@ -99,6 +102,42 @@ export function makeRouter(dependencies: ICaptureRouterDependencies) {
     }
   );
 
+  ////////////////////////////////////////////////////////////////////////////////
+  router.post(
+    '/initial-bodies',
+    bodyParser.json({ limit: '100mb' }),
+    async (req, res) => {
+      const { captureId } = req.params;
+      const { events, pathId, method } = req.body;
+      const manager = new InitialBodyManager();
+      const outputPaths = getInitialBodiesOutputPaths({
+        captureBaseDirectory: req.optic.paths.capturesPath,
+        captureId,
+        pathId,
+        method,
+      });
+      await fs.ensureDir(outputPaths.base);
+      await Promise.all([fs.writeJson(outputPaths.events, events)]);
```

Comment:
We moved away from spawning tasks right from API route handlers. Instead, we prefer all that logic to live somewhere inside the rest of the server process. One of the major reasons is that the task can very well outlive the request, but still needs to be properly managed. Even when that's not the case, it's nice to keep implementation details about the task, for the task.

With the diffs, we ended up writing a `Session` class that's attached to the `req` and exposes a method to start a new diff. https://github.com/opticdev/optic/blob/19386cab002eae2af7db3d8fe3969ce1248ff005/workspaces/cli-server/src/routers/capture-router.ts#L81-L100

Since the diff _is_ a longer running process, it then also exposes methods to query the results of the diff in   other route handlers.

https://github.com/opticdev/optic/blob/19386cab002eae2af7db3d8fe3969ce1248ff005/workspaces/cli-server/src/routers/capture-router.ts#L129-L139

Reply:
This is the only part I'd call out-- the response is the thing managing the lifecycle here. You start a request, it does work, it returns the result to the UX. The constract, as currently written does not have an affordance for outliving the HTTP call

Reply:
Those lifecycles match and are tied, point taken. That leaves a bunch of other concerns that don't really belong in a route handler, like setting up the config that's related to the session context, piping the logging and errors to the right place (which isn't the response, leaking implementation details we'll want to avoid), etc.

Reply:
Yeah, point 2, well taken. I'm checking out the new styles and taking a stab before happy hr.